src/petals/utils/peft.py

```diff
@@ -0,0 +1,105 @@
+import time
+from typing import List, Optional
+
+from hivemind.utils.logging import get_logger
+from huggingface_hub import HfFileSystem, get_hf_file_metadata, hf_hub_url
+from peft.utils import CONFIG_NAME, SAFETENSORS_WEIGHTS_NAME, PeftConfig
+from safetensors import safe_open
+from safetensors.torch import load_file
+from transformers.utils import get_file_from_repo
+
+from petals.utils.disk_cache import allow_cache_reads, allow_cache_writes, free_disk_space_for
+
+logger = get_logger(__name__)
+
+
+def check_peft_repository(repo_id: str) -> bool:
+    fs = HfFileSystem()
+    list_of_files = fs.glob(f"{repo_id}/{SAFETENSORS_WEIGHTS_NAME}", detail=False)
+    return len(list_of_files) > 0
+
+
+def load_specific_module(layers_name: List[str], filepath: str, framework: str = "pt"):
+    tensors = dict()
+    is_tensors_found = dict()
+    with safe_open(filepath, framework=framework) as f:
+        for k in f.keys():
+            for layer_name in layers_name:
+                if k.startswith(layer_name):
+                    is_tensors_found[layer_name] = True
+                    tensors[k] = f.get_tensor(k)
+        for layer_name in layers_name:
+            if not is_tensors_found.get(layer_name, False):
+                logger.warning(f"There is no peft weights with prefix {layer_name}")
```

Comment:
I worry that the warning is not enough for this issue. Maybe I have to add an exception here?